### PR TITLE
Remove links from inside links in mediawiki writer

### PR DESCRIPTION
**Pull request by wgevaert**
Fixes issue https://github.com/jgm/pandoc/issues/8738

I am not sure if this is the most desired way to do it.

Other thing to consider:

```hs
removeLinks :: [Inline] -> [Inline]
removeLinks = walk go
 where
  go (Link _ ils _) = Underline ils
  go x = x
```

### CODE REVIEW

**Comment by jgm** [comment]
Looks good to me!  Maybe add a command test? See test/commands for the format.
Name it after the issue number.

File: test/command/8738.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+```
+% pandoc -f json -t mediawiki
```
Comment (jgm) [inline test/command/8738.md:2]: Instead of json, use native for the input (more compact and easily readable).

**Comment by wgevaert** [comment]
Anything else I have to do or can it be merged?

**Comment by jgm** [comment]
looks good!

**Pull request merged**